### ISSUE TITLE
Multi-word, camel case model relationship functions

### ISSUE THREAD

**Issue by zablockijj** [laravel/framework#14378]
In a model, I named a belongsToMany relation as such:

`    public function driverRoutes()
    {
        return $this->belongsToMany('App\Route', 'carpool_routes', 'carpool_id', 'route_id')
            ->where('pivot.user_role', 'driver')
            ->withPivot('is_mon', 'is_tue', 'is_wed', 'is_thu', 'is_fri', 'is_sat', 'is_sun', 'date_departure', 'date_arrival', 'status', 'score', 'sort', 'user_role')
            ->withTimestamps();
    }`

When I attempt to access the property using lazy loading as:

`$carpool->driverRoutes`

I get a null value. However, if I change the name to a single all-lowercase word or change both to snakecase, it loads data from the database. I am wondering if I shouldn't be able to name and access relationships like this using camel case naming.

**Comment by GrahamCampbell** [comment]
We only support snake_case for relation names.

**Comment by josemiguelq** [comment]
@zablockijj 

Network.php
```php
  public function pointsOfSale()
    {
        return $this->hasMany(PointOfSale::class);
    }
PointOf
```
Sale.php
```php
 public function network()
    {
        return $this->belongsTo(Network::class, 'networkId');
    }
```

**Comment by senyahnoj** [comment]
According to https://laravel.com/docs/5.8/contributions laravel supports the PSR-2 coding standard. You are only allowed camel case method names in PSR-2.

This is a pain in projects which enforce PSR-2 with things like codesniffer.

**Comment by LucaRed** [comment]
@senyahnoj: it looks like camel case relationship methods now work fine on Laravel 5.8 :+1:

**Comment by trip-somers** [comment]
We've been looking at this for a bit, too, as we try to decide whether or not to make an exception to our PSR-1/PSR-12 standard for method names to allow snake_case() relationship method names.

Our reason for this is that we want to be able to access relationships as if they are attributes, and all of our multi-word database fields are snake_case; so something like `$order->line_items` is preferable to us in both PHP and in front-end javascript as `order.line_items`.

This is in order to create consistency when coding, such that developers -- both back-end and front-end -- can consistently handle everything as snake_case instead of having to not only go back and forth. For example, a fully hydrated and serialized Order might have a chain that looks like this: `$order->lineItems->item->package_weight` or `order.lineItems.item.package_weight`.

We are on board with the camelCase naming standard, so here's where we got walled in:

Is there a reason that you can use an accessor mutator via snake_case or camelCase while a relationship pseudo-attribute _must_ match the relationship method name exactly?

In other words, it would be great to be able to do the following:

```php
// defined as....
class Order extends Model
{
    public function lineItems()
    {
        return $this->hasMany(LineItem::class);
    }

   public function getSortedLineItemsAttribute($value)
    {
        return $this->lineItems()->orderBy('name')->get();
    }
}

// and accessed via....
$order->line_items;

// because either of these works...
$order->sorted_line_items;
$order->sortedLineItems;
```
